lib/std/fmt/ryu128.zig

```diff
@@ -0,0 +1,1117 @@
+//! This file implements the ryu floating point conversion algorithm:
+//! https://dl.acm.org/doi/pdf/10.1145/3360595
+
+const std = @import("std");
+
+const special_exponent = 0x7fffffff;
+
+/// Any buffer used for `format` must be at least this large. This is asserted. A runtime check will
+/// additionally be performed if more bytes are required.
+pub const min_buffer_size = 53;
+
+/// Returns the minimum buffer size needed to print every float of a specific type and format.
+pub fn bufferSize(comptime mode: Format, comptime T: type) comptime_int {
```

Comment:
This needs to be public if we actually want external users to be able take advantage of this. I originally preferred the direct buffer write approach instead of using a fmt writer.

Typically though, a user can get away with much less and these are only a concern for unbounded full-precision decimal output.